src/kOS/Suffixed/ResourceTransferValue.cs

```diff
@@ -0,0 +1,387 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using kOS.Execution;
+using kOS.Safe.Encapsulation;
+using kOS.Safe.Encapsulation.Suffixes;
+using kOS.Safe.Utilities;
+using kOS.Suffixed.Part;
+using Math = System.Math;
+
+namespace kOS.Suffixed
+{
+    public class ResourceTransferValue : Structure
+    {
+        private const float RESOURCE_SHARE_PER_UPDATE = 0.005f;
+        private enum TransferPartType
+        {
+            Part,
+            Parts,
+            Element
+        }
+
+        private readonly double? amount;
+        private double transferedAmount;
+        private readonly TransferManager transferManager;
+        private readonly PartResourceDefinition resourceInfo;
+        private readonly object transferTo;
+        private TransferPartType transferToType;
+        private readonly object transferFrom;
+        private TransferPartType transferFromType;
+        private TransferManager.TransferStatus status;
+
+        public string StatusMessage { get; private set; }
+
+        public TransferManager.TransferStatus Status
+        {
+            get { return status; }
+            private set
+            {
+                status = value;
+                transferManager.ReregisterTransfer(this);
+            }
+        }
+
+        public ResourceTransferValue(TransferManager transferManager, PartResourceDefinition resourceInfo, object transferTo, object transferFrom, double amount) :this (transferManager, resourceInfo, transferTo, transferFrom)
+        {
+            this.amount = amount;
+        }
+
+        public ResourceTransferValue(TransferManager transferManager, PartResourceDefinition resourceInfo, object transferTo, object transferFrom)
+        {
+            this.transferManager = transferManager;
+            this.resourceInfo = resourceInfo;
+            this.transferTo = transferTo;
+            this.transferFrom = transferFrom;
+            Status = TransferManager.TransferStatus.Inactive;
+
+            DetermineTypes();
+            InitializeSuffixes();
+        }
+
+        public void Update(double deltaTime)
+        {
+            if (Status != TransferManager.TransferStatus.Transferring) { return; }
+
+            IList<global::Part> fromParts = GetParts(transferFromType, transferFrom);
+            SafeHouse.Logger.Log("TRANSFER: FromParts Count: " + fromParts.Count);
```

Comment:
because i forgot to mark them as TODO:Remove i just spaced that they were there. I think just one when its starts and another when it fails or succeeds should be fine.